.github/workflows/unit_tests.yaml

```diff
@@ -0,0 +1,34 @@
+---
+name: Unit tests
+"on":
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+jobs:
+  go-test:
+    runs-on: ubuntu-latest
+    container:
+      image: registry.access.redhat.com/ubi8/go-toolset:1.19.13-2.1698062273
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Install jq
+        run: |
+          mkdir -p $HOME/packages && \
+          curl -L \
+          https://github.com/stedolan/jq/releases/download/jq-1.7/jq-linux64 \
+          -o $HOME/packages/jq && \
+          chmod +x $HOME/packages/jq
+      - name: Cache jq directory
+        id: cache-jq
+        uses: actions/cache@v2
+        with:
+          path: $HOME/packages/jq
+          key: jq-${{ runner.os }}-${{ hashFiles('**/jq') }}
```

Comment:
you can't use the files you are actually caching as the cache key, that will not allow calculating the key before performing the "Install jq" step that the cache is supposed to prevent (And isn't ATM).

Reply:
simplified

Reply:
update: There seems to be a weird issue with the caching
I added an extra-verification to see the path exists after installing jq, and it does.
yet the caching still fails:
```
/usr/bin/docker exec  92472cb68d90aa0998a0427e7f376de148d6bdb0e4ff24f8c6b4f39e50fd4720 sh -c "cat /etc/*release | grep ^ID"
Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.
```

Reply:
Try using `~` instead of `$HOME`, like in the examples here: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows

Also, you need to have the step that is fetching the data with curl happen *after* the step that is trying to restore it from cache so it can be skipped with an `if` instruction of the data is already cached.

Reply:
Thanks, this and some more changes fixed the issue, all is working well